src/datatypes/src/data_type.rs

```diff
@@ -161,6 +161,7 @@ impl ConcreteDataType {
                 | ConcreteDataType::Interval(_)
                 | ConcreteDataType::Duration(_)
                 | ConcreteDataType::Decimal128(_)
+                | ConcreteDataType::Binary(_)
```

Comment:
What if the data in binary isn't encoded in UTF-8?

Reply:
Ah sry, I didn't think about this. We can convert strings into binary values, but not the same in turn.

Here `parse_string_to_value` checks the conversion from other types into strings for converting strings into other types,  maybe we need to change this?
https://github.com/GreptimeTeam/greptimedb/blob/77904adaaf9c970fc7487f62480e7fa737090baf/src/sql/src/statements.rs#L61-L68

Also, I noticed that MySQL displays binary values like `0x313233` without caring about its encoding. I wonder if its only about displaying or we could take it as an option for the conversion between strings and blobs.

Reply:
`is_stringifiable()` is different from `is_string()` so this change should be safe. Maybe we should document that the method is used to indicate that the type could be parsed from a string literal.